libsolidity/interface/CompilerStack.cpp

```diff
@@ -1024,6 +1025,36 @@ Json::Value CompilerStack::methodIdentifiers(string const& _contractName) const
 	return methodIdentifiers;
 }
 
+Json::Value CompilerStack::errorIdentifiers(string const& _contractName) const
+{
+	if (m_stackState < AnalysisPerformed)
+		solThrow(CompilerError, "Analysis was not successful.");
+
+	Json::Value errorIdentifiers(Json::objectValue);
+	for (auto const& it: contractDefinition(_contractName).interfaceErrors())
+	{
+		string signature = it->functionType(true)->externalSignature();
+		errorIdentifiers[signature] = toHex(toCompactBigEndian(selectorFromSignature32(signature), 4));
+	}
+
+	return errorIdentifiers;
+}
+
+Json::Value CompilerStack::eventIdentifiers(string const& _contractName) const
+{
+	if (m_stackState < AnalysisPerformed)
+		solThrow(CompilerError, "Analysis was not successful.");
+
+	Json::Value eventIdentifiers(Json::objectValue);
+	for (auto const& it: contractDefinition(_contractName).interfaceEvents())
+	{
+		string signature = it->functionType(true)->externalSignature();
+		eventIdentifiers[signature] = toHex(toCompactBigEndian(selectorFromSignature32(signature), 4));
```

Comment:
There is one issue here. There are two kinds of events, the regular events and anonymous events. For the regular event, the `topic` for the `log*` opcode is the full 32 byte `keccak` hash of the selector. For anonymous events, this is skipped. So there are two things to be changed here:

1. Skip the signature for anonymous events. Check for `EventDefinition::isAnonymous()`.
2. For the regular ones, the signature should be the full 32 bytes. Here's an example from the codebase, where it is computed

https://github.com/ethereum/solidity/blob/a07b3ec70f91fdbf35929d5d474e91e739f792e8/libsolidity/codegen/ir/IRGeneratorForStatements.cpp#L998